src/cdk/app.ts

```diff
@@ -4,13 +4,40 @@ import { MiraServiceStack, MiraStack } from './stack'
 import { Stack } from '@aws-cdk/core'
 import * as fs from 'fs'
 import * as path from 'path'
-import { getBaseStackName, getBaseStackNameFromParams } from './constructs/config/utils'
+import {
+  getBaseStackName,
+  getBaseStackNameFromParams
+} from './constructs/config/utils'
 import { MiraConfig } from '../config/mira-config'
 // eslint-disable-next-line
-const minimist = require('minimist')
+const minimist = require("minimist");
 
 const args = minimist(process.argv)
 
+interface Prototypable {
+  prototype: Stack
+}
+
+interface StackConstructable {
+  new(app: cdk.App, props: { env: string }): Stack
+}
+
+interface MiraStackConstructable {
+  new(service: MiraServiceStack): MiraStack
+}
+
+interface MiraServiceStackConstructable {
+  new(app: MiraApp): MiraServiceStack
+}
+
+export type MiraValidStack = Stack &
```

Comment:
I'm interested to know more about the unuary "&" operator.  Can we link to a doc for TS that describes what all is happening here?  I know my preference is to not rely on typing when it doesn't gain us anything.